discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,213 @@
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// metaLabelPrefix is the meta prefix used for all meta labels.
+// in this discovery.
+const (
+	metaLabelPrefix = model.MetaLabelPrefix + "scw_"
+	separator       = ","
+)
+
+// DefaultSDConfig is the default Scaleway Service Discovery configuration.
+var DefaultSDConfig = SDConfig{
+	Zone:            "fr-par-1",
+	Port:            80,
+	RefreshInterval: model.Duration(60 * time.Second),
+}
+
+type SDConfig struct {
+	// Project: The Scaleway Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple scaleway_sd_configs
+	Zone string `yaml:"zone"`
+	// Access Key
+	AccessKey string `yaml:"access_key"`
+	// Secret Key
+	SecretKey string `yaml:"secret_key"`
+	// FilterName to filter on during the ListServers
+	FilterName string `yaml:"name"`
+	// FilterTags to filter on during the ListServers
+	FilterTags []string `yaml:"tags"`
+
+	HTTPClientConfig config.HTTPClientConfig `yaml:",inline"`
+
+	RefreshInterval model.Duration `yaml:"refresh_interval"`
+	Port            int            `yaml:"port"`
+}
+
+func (c SDConfig) Name() string {
+	return "scaleway"
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultSDConfig
+	type plain SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	return c.HTTPClientConfig.Validate()
+}
+
+func (c SDConfig) NewDiscoverer(options discovery.DiscovererOptions) (discovery.Discoverer, error) {
+	return NewDiscovery(&c, options.Logger)
+}
+
+// SetDirectory joins any relative file paths with dir.
+func (c *SDConfig) SetDirectory(dir string) {
+	c.HTTPClientConfig.SetDirectory(dir)
+}
+
+func init() {
+	discovery.RegisterConfig(&SDConfig{})
+}
+
+// Discovery periodically performs Scaleway requests. It implements
+// the Discoverer interface.
+type Discovery struct {
+	*refresh.Discovery
+	client    *scw.Client
+	port      int
+	zone      string
+	project   string
+	accessKey string
+	secretKey string
+	name      string
+	tags      []string
+}
+
+func NewDiscovery(conf *SDConfig, logger log.Logger) (*Discovery, error) {
+	d := &Discovery{
+		port:      conf.Port,
+		zone:      conf.Zone,
+		project:   conf.Project,
+		accessKey: conf.AccessKey,
+		secretKey: conf.SecretKey,
+		name:      conf.FilterName,
+		tags:      conf.FilterTags,
+	}
+
+	rt, err := config.NewRoundTripperFromConfig(conf.HTTPClientConfig, "scaleway_sd", false, false)
+	if err != nil {
+		return nil, err
+	}
+
+	profile, err := d.loadProfile()
```

Comment:
Can we avoid this step ? I'd like to have everything defined in the prometheus config. Currently that seems possible.